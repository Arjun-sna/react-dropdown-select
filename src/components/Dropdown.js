import React from 'react';
import styled from '@emotion/styled';

import NoData from './NoData';
import Item from './Item';

const Dropdown = ({ parentProps, parentState, parentMethods }) => (
  <DropDown
    tabIndex="-1"
    aria-expanded="true"
    role="list"
    selectBounds={parentState.selectBounds}
    dropdownGap={parentProps.dropdownGap}
    className="react-dropdown-select-dropdown">
    {parentProps.dropdownRenderer ? (
      parentProps.dropdownRenderer(parentProps, parentState, parentMethods)
    ) : (
      <React.Fragment>
        {parentMethods.searchResults().length === 0 ? (
          <NoData
            className="react-dropdown-select-no-data"
            parentState={parentState}
            parentProps={parentProps}
            parentMethods={parentMethods}
          />
        ) : (
          parentMethods
            .searchResults()
            .map((item, itemIndex) => (
              <Item
                key={item[parentProps.valueField]}
                item={item}
                itemIndex={itemIndex}
                parentState={parentState}
                parentProps={parentProps}
                parentMethods={parentMethods}
              />
            ))
        )}
      </React.Fragment>
    )}
  </DropDown>
);

Dropdown.propTypes = {};

const DropDown = styled.div`
  position: absolute;
  top: ${({ selectBounds, dropdownGap }) => selectBounds.height + 2 + dropdownGap}px;
  left: 0;
  border: 1px solid #ccc;
  width: ${({ selectBounds }) => selectBounds.width}px;
  padding: 0;
  display: flex;
  flex-direction: column;
  background: #fff;
  border-radius: 2px;
  box-shadow: 0 0 10px 0 #0000003b;
  max-height: 300px;
  overflow: auto;
  
  :focus {
    outline: none;
  }
}
`;

export default Dropdown;
