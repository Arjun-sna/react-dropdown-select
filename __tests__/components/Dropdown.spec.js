import React from 'react';
import renderer from 'react-test-renderer';

import Dropdown from '../../src/components/Dropdown';

const props = {
  parentProps: {
    dropdownRenderer: null,
  },
  parentState: {
    selectBounds: {}
  },
  parentMethods: {
    searchResults: () => []
  }
};

it('<Dropdown/> renders correctly', () => {
  const tree = renderer.create(<Dropdown {...props}/>).toJSON();

  expect(tree).toMatchSnapshot();
});
