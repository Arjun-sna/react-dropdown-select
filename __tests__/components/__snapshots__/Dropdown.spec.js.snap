// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dropdown/> renders correctly 1`] = `
<div
  aria-expanded="true"
  className="react-dropdown-select-dropdown react-dropdown-select-dropdown-position-undefined css-16z139g-DropDown e1qjn9k90"
  role="list"
  tabIndex="-1"
>
  <div
    className="react-dropdown-select-no-data css-1lli8k7-NoDataComponent e1l5ho1t0"
  />
</div>
`;
